### PR TITLE
Clean up the core grains tests

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### CODE REVIEW

**Review by Ch3LL — APPROVED** [review]

File: tests/pytests/unit/grains/test_core.py [hunk 1]
```diff
@@ -845,13 +852,14 @@ def test_almalinux_8_os_grains(os_release_dir):
 
 def test_unicode_error():
     raise_unicode_mock = MagicMock(name="raise_unicode_error", side_effect=UnicodeError)
-    with patch("salt.grains.core.hostname"):
-        with patch("socket.getaddrinfo", raise_unicode_mock):
-            ret = salt.grains.core.ip_fqdn()
-            assert ret["fqdn_ip4"] == ret["fqdn_ip6"] == []
+    with patch("salt.grains.core.hostname"), patch(
+        "socket.getaddrinfo", raise_unicode_mock
```
Comment (waynew) [inline tests/pytests/unit/grains/test_core.py:856]: Well, I see this has been merged now, but I was going to suggest that this be `patch('socket.getaddrinfo', side_effect=UnicodeError)` :upside_down_face:

File: tests/pytests/unit/grains/test_core.py [hunk 2]
```diff
@@ -1107,25 +1116,28 @@ def test__windows_os_release_grain():
...
+            assert version == expected_version
 
     # Special Cases
     # Windows Embedded Standard is Windows 7
     caption = "Windows Embedded Standard"
-    with patch("platform.release", MagicMock(return_value="7")):
-        version = core._windows_os_release_grain(caption, 1)
-        assert version == "7"
+    with subtests.test(caption):
+        with patch("platform.release", MagicMock(return_value="7")):
```
Comment (waynew) [inline tests/pytests/unit/grains/test_core.py:1128]: **suggestion** similar simplification here - no need to create a new MagicMock, just `, return_value=7)`

File: tests/pytests/unit/grains/test_core.py [hunk 3]
```diff
@@ -1174,160 +1182,154 @@ def test_bsd_memdata():
...
-                    with patch.dict(core.__salt__, {"cmd.run_all": MagicMock()}):
-                        grains = core._virtual({"kernel": "Linux"})
-                        assert grains.get("virtual_subtype") == "Docker"
-                        assert grains.get("virtual") == "container"
+    cgroup_data = "10:memory{}a_long_sha256sum".format(cgroup_substr)
+    log.debug("Testing Docker cgroup substring '%s'", cgroup_substr)
+    with patch.object(os.path, "isdir", MagicMock(return_value=False)), patch.object(
+        os.path,
+        "isfile",
+        MagicMock(side_effect=lambda x: True if x == "/proc/1/cgroup" else False),
```
Comment (waynew) [inline tests/pytests/unit/grains/test_core.py:1238]: **suggestion** another set of extra MagicMock declarations (the run_all one is valid though)

**Review by waynew — COMMENTED** [review]
A bit late to the party - several of these could be simplified even further.

**Pull request merged**